app/js/enhancedDate.js

```diff
@@ -1,5 +1,94 @@
 /* 
  * add your module definition for enhancedDate here
  * assign the module to a global variable
- * /
+ */
+
+var enhancedDate = (function enhancedDate() {
+
+    'use strict';
+ 		 
+	var date = null;
+	var isDateSet = false;
+	var days = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 
+			'Friday', 'Saturday'];
+	var months = ['Januray', 'February', 'March', 'April', 'May', 'June',
+			'July', 'August', 'September', 'November', 'December'];
+
+    var publicAPI = {
+
+    	setDate: function setDate(myDate) {
+
+    		if (myDate == "Invalid Date") {
+    			alert("The date you've entered is invalid.");
+    		} else if (myDate instanceof Date) {
+				date = myDate;
```

Comment:
This works, but keep in mind that objects are pass by reference when used as arguments in JavaScript. This means that if the object that was passed in is modified outside the module, it will update the reference you're storing internally. Instead, just instantiate a new Date reference by passing in the date the user passed in: `date = new Date(myDate);`